src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.cs

```diff
@@ -509,7 +563,20 @@ private static bool ComputeNeedsValueContentAnalysis(IOperation operationBlock,
                         var parentAttributes = CopyAttributes(attributes);
                         using var parentCapturedVersions = PooledDictionary<string, Version>.GetInstance(capturedVersions);
 
-                        if (!IsKnownValueGuarded(parentAttributes, ref csAttributes, parent, parentCapturedVersions))
+                        if (value.AnalysisValues.Count != 0 && value.AnalysisValues.Count == parent.AnalysisValues.Count)
+                        {
+                            var parentEnumerator = parent.AnalysisValues.GetEnumerator();
+                            foreach (var val in value.AnalysisValues)
+                            {
+                                if (parentEnumerator.MoveNext() && !parentEnumerator.Current.Equals(val.GetNegatedValue()))
+                                {
+                                    goto label;
+                                }
+                            }
+                            continue;
+                        }
+
```

Comment:
This is for merging `OperatingSystem.IsIOS() && !OperatingSystem.IsMacCatalyst()` conditional which flow analysis value is `(OperatingSystem.IsIOS() || OperatingSystem.IsMacCatalyst()) && !OperatingSystem.IsMacCatalyst()`. Ideally it should be handled/merged by the flow analisys into just `OperatingSystem.IsIOS()` CC @mavasani